### DISCUSSION TITLE
TShock 5.1.2 for Terraria 1.4.4.8.1 (ζ Ori)

### DISCUSSION THREAD

**Discussion by hakusaro** [Pryaxis/TShock#2826] [Announcements]
Welcome to TShock for Terraria 5.1.2 with support for Terraria 1.4.4.8.1. This is, of course, also on nuget.

**Before installing, make a complete backup of your existing server and all files and folders, including `tshock.sqlite` and the `tshock` folder. We are not responsible for failed installations. Please make sure you backup your data. Severe data loss or gain may occur.**

## Notable changes in this release
* Added support for Terraria 1.4.4.8.1 via OTAPI 3.1.19. (@SignatureBeef)

## Notable changes in the last release
* Fixed item giving potentially dropping too many items. (@PotatoCider, @punchready)
* Excluded GeoIP.dat from release bundle. (@SignatureBeef)
* Added `TownSlimeRed` to `FishableNpcIDs` list, allowing it to be fished up. (@drunderscore)
* Bump to Terraria 1.4.4.8 via OTAPI 3.1.18. (@hakusaro, @SignatureBeef)
  * In this version of Terraria, `Main.maxBuffTypes` and other `maxWhateverTypes` fields have been removed. Their replacements are in `Terraria.ID.whateverID.Count`. TShock calls to these fields have been swapped in order to bring forward compatibility with Terraria 1.4.4.8.
* In OTAPI 3.1.17, allowed Crystal Shard to grow. (@sgkoishi, @cc004, SignatureBeef/Open-Terraria-API#96)
* Added permission for summoning Mechdusa, Deerclops and slime pet. (@sgkoishi, #2808)
* Changed login to only restrict CC'd players during login whilst SSC is enabled. (@drunderscore)
  * This change allows the config option `RequireLogin` to function usefully again when SSC is not enabled.
* Changed `PlayerData.RestoreCharacter` to remove all buffs. (@drunderscore)
  * Before this change, it was theoretically possible to smuggle buffs onto servers with SSC enabled, by using buff indexes past `22`.
* Allowed Torch God's Favor to place different types of torches and campfires. (@sgkoishi, #2811)
* Updated translations! Currently, the major projects are at the following completion rates:
  * Chinese (93%)
  * Portuguese, Brazilian (89%)
  * Indonesian (89%)
  * Russian (56%)
  * Spanish (24%)
  * toki pona (10%)
  * Turkish (8%)
  * For complete credits, see the table below.

### Translation credits
The following translators contributed changes from November 1, 2022 to November 9, 2022 (from TShock 5.0.0 to 5.1.0).

|Contributor | Language | Translated words|
|-----|-----|-----|
|Shiva Goddess (ShivaGoddess) | Portuguese, Brazilian | 6979|
|Ricko (Rickojp) | Portuguese, Brazilian | 3154|
|RidnRaven (ridwankun2) | Indonesian | 2329|
|Janet Blackquill (pontaoski) | toki pona | 1216|
|SGKoishi | Chinese Simplified | 640|
|Cristofer GamerTVH (cristoferherame) | Spanish | 622|
|HDSeventh (hdseventh) | Indonesian | 315|
|EMRE ÇELİK (emre0447) | Turkish | 312|
|PHPoenX | Russian | 297|
|./lemon.sh (lemon-sh) | Polish | 206|
|Сергей Червяков (chsergeyg) | Russian | 182|
|okaythisisepic | Russian | 80|
|KomashiOFC | Portuguese, Brazilian | 76|
|Runesicle | toki pona | 22|
|Marotheit | Pirate English | 7|
|ATFGK | Chinese Simplified | 4|

## Upgrading

**Before installing, make a complete backup of your existing server and all files and folders, including `tshock.sqlite` and the `tshock` folder. We are not responsible for failed installations. Please make sure you backup your data. Severe data loss or gain may occur.**

To upgrade TShock, replace all of the files in the location of your server installation with all of the files from the release zip we provide. Overwrite all files. No data will be lost.

<hr /><em>This discussion was created from the release <a href='https://github.com/Pryaxis/TShock/releases/tag/v5.1.2'>TShock 5.1.2 for Terraria 1.4.4.8.1 (ζ Ori)</a>.</em>

**Comment by SirSaiph** [comment]
This needs an update, Terraria is 1.4.4.9 now.

**Comment by lyavkube** [comment]
There is updates by updates NotLikeThis

**Reply by SirSaiph** [reply]
English please?

**Reply by lyavkube** [reply]
nope

**Comment by Logz11** [comment]
Does anyone have any tips for running this in an AWS EC2 Ubuntu 22.04 LTS instance? I've been [following this guide](https://levelup.gitconnected.com/guide-building-a-terraria-server-on-aws-eaebd6d556a5) but am stuck at the command `cd ~/Terraria/TShock4.5.17_Terraria_1.4.3.6` as the new linux x64.zip is not a directory upon unzipping.

I ran the command `tar -xvf TShock-Beta-linux-x64-Release.tar` but didn't get a .exe file. I've tried executing `./TShock.Server` in the terminal to no avail.

Does anyone know if I'm using the wrong download, or have missed a step?

Would greatly appreciate if anyone has any guidance on this issue! Thank you for your time!

**Reply by hakusaro** [reply]
Did you install the .NET runtime? You have to do that before TShock.Server will execute.

**Reply by Logz11** [reply]
Ah, I don't think my server has that. I'll download it and give it a shot. Thank you!

**Comment by aarvndhNG** [comment]
What is **maxWhateverTypes** in tshock 5.1.3

**Reply by punchready** [reply]
`Main.maxNPCTypes` etc. are Terraria fields and have nothing to do with TShock. In 1.4.4 these consts have been moved to their respective `ID` classes, such as `NPCID.Count` in this example.
